cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/TimePartitionedFileSetDataset.java

```diff
@@ -149,6 +149,7 @@ public TimePartitionOutput getPartitionOutput(long time) {
         "Output is not supported for external time-partitioned file set '" + spec.getName() + "'");
     }
     PartitionKey key = partitionKeyForTime(time);
+    assertNotExists(key);
```

Comment:
Actually, if a user calls `PartitionedFileSet#getOutputPartition` outside of a transaction, we can `LOG.warn` and check the file system for partition existence.